### PR TITLE
feat(terraform) Remove parameter `agentContainerImage` in favor of `agentLabel`

### PR DESCRIPTION
**Pull request by smerle33**
Change in the pipeline shared library to start using labelled arm64 agent instead of PodTemplate
as per https://github.com/jenkins-infra/helpdesk/issues/3823#issuecomment-1873857618

### CODE REVIEW

**Review by dduportal — COMMENTED** [review]
Almost there, but more work is required:

- Title need to be updated
- Need to update documentation
- Need to run a fully fledged test with a PR on one of the terraform jobs

**Comment by smerle33** [comment]
need to update https://github.com/jenkins-infra/shared-tools/tree/main/terraform#jenkins-pipeline to document changes

**Comment by smerle33** [comment]
using this PR to test on AZURE : https://infra.ci.jenkins.io/job/terraform-jobs/job/azure/job/PR-575

File: vars/terraform.groovy [hunk 1]
```diff
@@ -59,6 +59,9 @@ def call(userConfig = [:]) {
               }
               if (finalConfig.runCommonTests) {
                 stage('✅ Commons Test Terraform Project') {
+                  sh 'pwd'
+                  sh "echo ${PATH}"
```
Comment (dduportal) [inline vars/terraform.groovy:63]:
```suggestion
                  sh 'echo ${PATH}'
```

**Comment by smerle33** [comment]
followed by https://github.com/jenkins-infra/shared-tools/pull/136 to update documentation

**Review by dduportal — APPROVED** [review]

**Pull request merged**